docs/activity.md

```diff
@@ -0,0 +1,241 @@
+let's do that by adding by this line at the end of the function :
+
+```python
+bird.warn_about_cat()
+```
+
+[Solution](https://github.com/microbit-carlos/microbit-bird-activity/tree/main/solutions/2-cat.md)
+
+
+## Bird state
+
+We've learned how to react to environment events (received via micro:bit
+radio), what about things that happen directory to our bird?
+
+Let's look further down `main.py` and we'll find this infinite loop.
+
+```python
+while True:
+    # Look inside yourself and listen, how is your bird feeling?
+    bird_state = bird.current_state()
+    if bird_state == "chill":
+        display.show(Image.HAPPY)
+    sleep(100)
+```
+
+The function `bird.current_state()` uses the micro:bit on board sensors to
+figure how your bird is doing.
+
+<p class="exercise">
+Exercise: <br>
+Birds don't like to be shaken, so "angry" can also be one of the bird states
+to code.
+<br>
+Let's expand that "if" statement with an "elif" and do something else when
+the bird is "angry".
+</p>
+
+[Solution](https://github.com/microbit-carlos/microbit-bird-activity/tree/main/solutions/3-state-angry.md)
+
+
+## SoundEffects
+
+TODO: Need to explain this, and make it appealing for users to spend most of
+the time here.
+
+```python
+audio.play(audio.SoundEffect(
+    freq_start=400,
+    freq_end=900,
+    duration=1000,
+    vol_start=255,
+    vol_end=0,
+    wave=audio.SoundEffect.WAVE_NOISE,
+    fx=audio.SoundEffect.FX_TREMOLO,
+    interpolation=audio.SoundEffect.INTER_LINEAR
+))
+```
+
+The Python REPL is a great place to quickly try some Python code.
+
+To use the REPL, first connect to the micro:bit with the "Send to micro:bit"
+button. And then press the "Show serial" on the right of the dark box:
+
+![Hidden REPL screenshot](img/repl-hidden.png)
+
+Then the three dots menu on the right, and select the "Send Ctrl+C" option:
+
+<img src="/microbit-bird-activity/img/repl-menu.png" alt="REPL menu" width="200px">
+
+And there we go, the MicroPython REPL, where you can type or paste any Python
+code you like, and it will execute it inmediately:
+
+![Hidden REPL screenshot](img/repl-expanded.png)
+
+<p class="exercise">
+Exercise: <br>
+Can you come up with an "angry" sound for the bird state?
+What kind of sound would a bird make if they spot a cat?
+</p>
+
+There is no "solution" for this exercise, be as creative as you'd like!
+
+Have you tried playing multiple sounds one after another?
+You can make an ever longer cooler sound 📢. 
+
+```python
+effect_one = audio.SoundEffect(
+    freq_start=988,
+    freq_end=440,
+    duration=190,
+    vol_start=255,
+    vol_end=255,
+    wave=audio.SoundEffect.WAVE_SINE,
+    fx=audio.SoundEffect.FX_VIBRATO
+)
+effect_two = audio.SoundEffect(
+    freq_start=2570,
+    freq_end=440,
+    duration=874,
+    vol_start=255,
+    vol_end=87,
+    wave=audio.SoundEffect.WAVE_SAWTOOTH,
+    fx=audio.SoundEffect.FX_VIBRATO
+)
+
+audio.play(effect_one)
+audio.play(effect_two)
+```
+
+
+## More events, more sounds
+
```

Comment:
I wonder if this should be linked to from higher up in the doc? I found myself thinking "oh, can I just do those few things" in the earlier section and then felt happier when I got here.

Reply:
Yes, good call, added in 21ad6bda79cb278392ffcc3482dd4da27274a3cb